BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift

```diff
@@ -221,62 +211,61 @@ public final class ConvivaAnalytics: NSObject {
             return
         }
 
-        client.reportError(sessionKey, errorMessage: message, errorSeverity: severity)
+        videoAnalytics.reportPlaybackError(message, errorSeverity: severity)
         if endSession {
             internalEndSession()
         }
     }
 
     /**
-     Puts the session state in a notMonitored state.
+     Puts the session in a notMonitored state.
      */
-    public func pauseTracking() {
-        // AdStart is the preferred way to pause tracking according to conviva.
-        client.adStart(sessionKey,
-                       adStream: .ADSTREAM_SEPARATE,
-                       adPlayer: .ADPLAYER_SEPARATE,
-                       adPosition: .ADPOSITION_PREROLL)
-        client.detachPlayer(sessionKey)
+    public func pauseTracking(isBumper: Bool) {
+        self.isBumper = isBumper
+        let event: String = self.isBumper ?
+        CISConstants.getEventsStringValue(Events.BUMPER_VIDEO_STARTED) :
+        CISConstants.getEventsStringValue(Events.USER_WAIT_STARTED)
+        videoAnalytics.reportPlaybackEvent(event)
         logger.debugLog(message: "Tracking paused.")
     }
 
     /**
-     Puts the session state from a notMonitored state into the last one tracked.
+     Changes the session state from notMonitored state into the previous state.
      */
     public func resumeTracking() {
-        client.attachPlayer(sessionKey, playerStateManager: playerStateManager)
-        // AdEnd is the preferred way to resume tracking according to conviva.
-        client.adEnd(sessionKey)
+        let event: String = self.isBumper ?
```

Comment:
Shouldn't we reset `self.isBumper` again afterwards?

Reply:
It's not necessary in that resume tracking events without pause tracking first are ignored by Conviva. Given that we always set the bool on pauseTracking, this behaves correctly. But I've added it anyway for clarity :)